### PR TITLE
FEAT: [OmniSciDB] Add method parameter to load_data

### LINKED ISSUE #2164: FEAT: OmniScidb - Add "method" parameter to OmniSciDB load_table
Add to OmniSciDB load_data the paramter "method" (see https://pymapd.readthedocs.io/en/latest/api.html#pymapd.Connection.load_table)

method: string {‘infer’, ‘columnar’, ‘rows’, ‘arrow’}

Method to use for loading the data. Three options are available

- pyarrow and Apache Arrow loader
- columnar loader
- row-wise loader

The Arrow loader is typically the fastest, followed by the columnar loader, followed by the row-wise loader. If a DataFrame or pyarrow.Table is passed and pyarrow is installed, the Arrow-based loader will be used. If arrow isn’t available, the columnar loader is used. Finally, data is an iterable of tuples the row-wise loader is used.

### PR DESCRIPTION
**Pull request by xmnlab**
Resolve #2164

### CODE REVIEW

**Comment by xmnlab** [comment]
this PR is done for review. thanks!

File: ibis/omniscidb/tests/test_client.py
Comment (jreback) [inline ibis/omniscidb/tests/test_client.py:355]: is an arrow an explicit optional dependency of ominisc? can you update setup.py 
are we testing this on CI?

Comment (xmnlab) [reply ibis/omniscidb/tests/test_client.py:355]: I was checking the pymapd right now .. and it seems it is required now. thanks @jreback I will fix that now.

Comment (xmnlab) [reply ibis/omniscidb/tests/test_client.py:355]: also I opened an issue on pymapd related to pymapd.load_data docstring as now it seems pyarrow is required.

**Comment by xmnlab** [comment]
this PR is ready again for a new review.

**Comment by xmnlab** [comment]
a friendly reminder about this PR :)

**Comment by xmnlab** [comment]
a friendly reminder about this PR :) thanks

**Comment by datapythonista** [comment]
Release is conflicting.

**Comment by xmnlab** [comment]
thanks @datapythonista ! rebased!

File: ibis/omniscidb/tests/test_client.py
Comment (datapythonista) [reply ibis/omniscidb/tests/test_client.py]: @xmnlab I was checking this again. Can you add `pyarrow` to the `omniscidb_requires` in `setup.py` please? Since we're importing it, it's worth having the requirement explicit, even if it's a `pymapd` dependency.

**Comment by xmnlab** [comment]
branch rebased. thanks.

**Comment by datapythonista** [comment]
@xmnlab can you address this comment please? https://github.com/ibis-project/ibis/pull/2165#discussion_r460067702

**Comment by xmnlab** [comment]
@datapythonista sure. sorry I forgot that one. I am working on that. thanks

File: ibis/omniscidb/tests/test_client.py
Comment (xmnlab) [reply ibis/omniscidb/tests/test_client.py]: done!  thanks @datapythonista !

File: setup.py
Comment (jreback) [inline setup.py:25]: you would need to add an entry in ci/deps/omnsidb.yml

Comment (xmnlab) [reply setup.py:25]: make sense. I am going to add that! thanks

File: ibis/omniscidb/client.py
Comment (datapythonista) [inline ibis/omniscidb/client.py:1193]: Is removing `**kwargs` intentional?

File: ibis/omniscidb/tests/test_client.py
Comment (datapythonista) [inline ibis/omniscidb/tests/test_client.py:333]:
```suggestion
    pd.testing.assert_frame_equal(result, data)
```
I think it's clearer and more standard to use `result == expected` than the other way round.

File: ibis/omniscidb/tests/test_client.py
Comment (datapythonista) [inline ibis/omniscidb/tests/test_client.py:331]: I don't understand this, if you always want the pandas, we not simply use `assert_frame_equal(result, df_salary)` below instead?

File: ibis/omniscidb/tests/test_client.py
Comment (datapythonista) [inline ibis/omniscidb/tests/test_client.py:327]: I would prefer if you define `df_salary`, `arrow_salary`... inside the function, and you use the strings `pandas` and `arrow` for the `data` parameter (that can be renamed to `format`). Then you can do:
```suggestion
    con.load_data(temp_table, df_salary if format == 'pandas' else arrow_salary, method=method)
```
This will save memory in the testing job and will keep things more compact.

**Review by datapythonista — COMMENTED** [review]
Thanks @xmnlab for this. Looks good. You've got some conflicts, and I added few suggestions that I think will make the code better.

File: ibis/omniscidb/client.py
Comment (xmnlab) [reply ibis/omniscidb/client.py:1193]: yes, the initial idea for kwarg here was to allow a similar signature for all `load_data` functions across all the backends, and as it seems it is not necessary and it is not used inside the function, I am just removing that for now.

File: ibis/omniscidb/tests/test_client.py
Comment (xmnlab) [reply ibis/omniscidb/tests/test_client.py:331]: yes, as we are using the same data for both, it works. thanks

**Comment by xmnlab** [comment]
@datapythonista I applied the suggestion and CI is green. let me know if I missed anything here pls. Thanks!

File: ibis/backends/omniscidb/tests/test_client.py
Comment (xmnlab) [reply ibis/backends/omniscidb/tests/test_client.py:306]: yep ... it doesn't make any sense ... sorry ... no idea why it is in this format :X ... I will pay more attention next time ... also .. No idea neither why the tests passed :X 

but the way, the "matrix" of possibilities should be the same of the initial commit:

```
       ('rows', df_salary),
        ('columnar', df_salary),
        ('infer', df_salary),
        ('infer', arrow_salary),
        ('arrow', arrow_salary)
```

so it should be something like 

```
       ('rows', 'pandas'),
        ('columnar', 'pandas'),
        ('infer', 'pandas'),
        ('infer', 'arrow'),
        ('arrow', 'arrow')
```

I will fix this issue .. and sorry again for the noise ... I really don't know what happened

File: setup.cfg
Comment (xmnlab) [reply setup.cfg:19]: ok ... I will rebase the code again and try :) thanks!

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @xmnlab

**Comment by xmnlab** [comment]
@datapythonista , is there anything else I should address here? or can it be already merged? thanks

**Comment by datapythonista** [comment]
> @datapythonista , is there anything else I should address here? or can it be already merged? thanks

lgtm as I mentioned in the past comment. I'll let @jreback have a look before merging.

**Comment by datapythonista** [comment]
Sorry @xmnlab this conflicted once more. Can you merge master and fix the conflicts please? I'll merge as soon as this is green.

@jreback I think your review comments are all addressed. But if you want to have a look...

**Comment by xmnlab** [comment]
thanks @datapythonista ! branch rebased!

**Comment by datapythonista** [comment]
Thanks @xmnlab

**Pull request merged**
